tests/python/unittest/test_custom_datatypes_change_dtype.py

```diff
@@ -0,0 +1,481 @@
+    register_op(create_lower_func({
+        32: 'Posit32es2Log',
+        16: 'Posit16es2Log',
+        8: 'Posit8es2Log'
+    }), "Call", "llvm", "posites2", intrinsic_name="log")
+    register_op(create_lower_func({
+        32: 'Posit32es2Sigmoid',
+        16: 'Posit16es2Sigmoid',
+        8: 'Posit8es2Sigmoid'
+    }), "Call", "llvm", "posites2", intrinsic_name="sigmoid")
+    register_op(create_lower_func({
+        32: 'Posit32es2Tanh',
+        16: 'Posit16es2Tanh',
+        8: 'Posit8es2Tanh'
+    }), "Call", "llvm", "posites2", intrinsic_name="tanh")
+    register_min_func(lambda num_bits: - (2 ** 2 ** 2) ** (num_bits - 2), "posites2")
+
+    register("noptype", 132)
+    register_op(create_lower_func({
+        (32, 32): "FloatToNop32"
+    }), "Cast", "llvm", "noptype", "float")
+    register_op(create_lower_func({
+        (32, 32): 'Nop32ToFloat'
+    }), "Cast", "llvm", "float", "noptype")
+    register_op(create_lower_func({
+        (4, 32): "IntToNop32"
+    }), "Cast", "llvm", "noptype", "int")
+    register_op(create_lower_func({32: 'IntToNop32'}), "Add", "llvm", "noptype")
+    register_op(create_lower_func({32: 'Nop32Sub'}), "Sub", "llvm", "noptype")
+    register_op(create_lower_func({32: 'FloatToNop32'}), "FloatImm", "llvm",
+                "noptype")
+    register_op(create_lower_func({32: 'Nop32Mul'}), "Mul", "llvm", "noptype")
+    register_op(create_lower_func({32: 'Nop32Div'}), "Div", "llvm", "noptype")
+    register_op(create_lower_func({32: 'Nop32Max'}), "Max", "llvm", "noptype")
+    register_op(create_lower_func({32: 'Nop32Sqrt'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="sqrt")
+    # TODO(gus) not sure if this will work...
+    register_op(lower_ite,
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="tvm_if_then_else")
+    register_op(create_lower_func({32: 'Nop32Exp'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="exp")
+    register_op(create_lower_func({32: 'Nop32Log'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="log")
+    register_op(create_lower_func({32: 'Nop32Sigmoid'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="sigmoid")
+    register_op(create_lower_func({32: 'Nop32Tanh'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="tanh")
+    # This can be anything, considering the type isn't functionally correct.
+    register_min_func(lambda num_bits: 0, "noptype")
+
+
+def run_ops(src_dtype, dst_dtype, rtol=1e-7, atol=1e-7):
+    """Run the same op, but with two different datatypes"""
+    def check_unary_op(op, src_dtype, dst_dtype):
+        t1 = relay.TensorType((5, 10, 5), src_dtype)
```

Comment:
Chooses an arbitrary tensor shape. That's probably fine, but can we define the shape at the top of the function?

Reply:
yep done!